### PR TITLE
[PIM-6397] Search

### PR DESCRIPTION
**Pull request by pierallard**
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | y
| Added integration tests           | -
| Changelog updated                 | y
| Review and 2 GTM                  | Todo
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: features/Context/Page/Base/Grid.php [hunk 1]
```diff
@@ -310,12 +311,50 @@ public function checkOptionInFilter(array $expectedOptions, $filterName)
...
         $filter->open();
         $filter->filter($operator, $value);
     }
 
     /**
+     * @param string $value
+     */
+    public function search($value)
+    {
+        $this->spin(function () use ($value) {
```
Comment (juliensnz) [inline features/Context/Page/Base/Grid.php:328]: You should use our decorator system

File: features/Context/Page/Base/Grid.php [hunk 2]
```diff
@@ -310,12 +311,50 @@ public function checkOptionInFilter(array $expectedOptions, $filterName)
...
+    /**
+     * Returns the displayed criteria of a filter
+     *
+     * @param $filterName
+     *
+     * @return string
+     */
+    public function getCriteria($filterName)
+    {
+        $filter = $this->getFilter($filterName);
```
Comment (juliensnz) [inline features/Context/Page/Base/Grid.php]: same here: please use the decorator system

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/search-filter.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+        timer: null,
+
+        timerDelay: 500,
+
+        className: 'AknFilterBox-searchContainer filter-item search-filter',
+
+        render: function () {
+            this.$el.html(
+                '<input class="AknFilterBox-search" autocomplete="off" type="text" name="value" value="" placeholder="Search">'
```
Comment (juliensnz) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/search-filter.js]: please use a template file

Comment (juliensnz) [reply src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/search-filter.js]: and use a translation key for the place  holder

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/search-filter.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        /**
+         * @inheritDoc
+         */
+        _readDOMValue: function() {
+            return {
+                value: this._getInputValue(this.inputValueSelector)
+            };
+        },
+
+        keyPress: function(event) {
```
Comment (juliensnz) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/search-filter.js]: please add jsdoc

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/search-filter.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                '<input class="AknFilterBox-search" autocomplete="off" type="text" name="value" value="" placeholder="Search">'
+            );
+        },
+
+        /**
+         * @inheritDoc
+         */
+        _writeDOMValue: function(value) {
+            this._setInputValue(this.inputValueSelector, value.value);
+            return this;
```
Comment (juliensnz) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/search-filter.js]: please add blank line before return

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js [hunk 6]
```diff
@@ -30,11 +30,24 @@ function($, _, Backbone, mediator, MultiselectDecorator) {
...
+        },
+
+        /**
          * Container classes
          *
          * @property
          */
-        className: 'AknFilterBox filter-box oro-clearfix-width',
+        className: function () {
+            return (true === this.displayManageFilters())
```
Comment (juliensnz) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filters-manager.js]: why not:
```
return 'AknFilterBox filter-box oro-clearfix-width' + !this.displayManageFilters() ? 'AknFilterBox--search' : '';
```

**Review by juliensnz — CHANGES_REQUESTED** [review]

**Comment by juliensnz** [comment]
IMHO we should not merge this card: We try to trick the user by faking a full text search field. It will create confusion and will not fulfill properly any need. Like macron could say: "it some powder of perlimpimpin".

We should implement it properly and no try to fake something to make the PIM look good and shiny.

File: features/Context/Page/Base/Grid.php (hunk 1 above)
Comment (pierallard) [reply features/Context/Page/Base/Grid.php:328]: Done

**Review by a2xchip — APPROVED** [review]
GTM

**Comment by solivier** [comment]
agree with @juliensnz

File: features/Context/Page/Base/Grid.php (hunk 1 above)
Comment (juliensnz) [reply features/Context/Page/Base/Grid.php:328]: I was speaking about the filter stuff on the grid (ie: use filterBy method)

**Review by juliensnz — CHANGES_REQUESTED** [review]

**Comment by pierallard** [comment]
@juliensnz "I was speaking about the filter stuff on the grid (ie: use filterBy method)"
I can not use the lambda filter stuff because it does not fits with other filters actions (open the filter + fill operator + fill value + click on update).

**Pull request closed without merging**